### PR TITLE
GH1383: Add casts to ERR_PACK

### PR DESCRIPTION
**Pull request by richsalz**
And update script to not re-introduce whitespace glitch.

### CODE REVIEW

File: include/openssl/err.h [hunk 1]
```diff
@@ -133,10 +133,12 @@ typedef struct err_state_st {
...
 
-# define ERR_PACK(l,f,r) \
-  ( ((unsigned int)((l) & 0x0FF) << 24L) | (((f) & 0xFFF) << 12L) | ((r) & 0xFFF) )
-# define ERR_GET_LIB(l)          (int)((((unsigned long)l)>>24L)&0xffL)
-# define ERR_GET_FUNC(l)         (int)((((unsigned long)l)>>12L)&0xfffL)
+# define ERR_PACK(l,f,r) ( \
+        (((unsigned int)(l) & 0x0FF) << 24L) | \
+        (((unsigned int)(f) & 0xFFF) << 12L) | \
+        (((unsigned int)(r) & 0xFFF)       ) )
+# define ERR_GET_LIB(l)          (int)((((unsigned long)l) >> 24L) & 0x0FFL)
```
Comment (ekasper) [inline include/openssl/err.h]: You might as well cast these to `unsigned int` as well for consistency. Does `ERR_GET_REASON` also need a cast then?

**Comment by ekasper** [comment]
+1 but see comment

File: include/openssl/err.h (hunk 1 above)
Comment (richsalz) [inline include/openssl/err.h]: you mean liens 140,141,142?  If so yes, that makes sense.

Comment (richsalz) [inline include/openssl/err.h]: Assuming I got it right, I pushed an additional commit.  Please take a look?

Comment (ekasper) [inline include/openssl/err.h]: Um, I meant unifying all casts on the input to silence the warnings:

a) `unsigned long` -> `unsigned int`; and
b) a cast on the input of `ERR_GET_REASON`, similarly to the others.

We shouldn't change the return type.

Comment (richsalz) [inline include/openssl/err.h]: Doh!  Fixed.  Revised commit pushed.

Comment (ekasper) [inline include/openssl/err.h]: Hmm, I looked at the docs meanwhile - `ERR_PACK` is documented to take in `int`s, so these three casts are needed. But the `ERR_GET_` flavours take an `unsigned long` so actually I think these input casts should simply be removed.

(Sorry for making you go around in circles.)

Comment (richsalz) [inline include/openssl/err.h]: No prob.  I like getting dizzy :)  updated commit pushed.

**Pull request closed without merging**